react_juce/core/TextInputView.h

```diff
@@ -0,0 +1,120 @@
+/*
+==============================================================================
+
+  TextInputView.h
+  Created: 20 Jan 2021 10:30pm
+
+==============================================================================
+*/
+
+#pragma once
+
+#include "View.h"
+
+namespace blueprint
+{
+    //==============================================================================
+    /** The TextInput class is an extended juce::TextEditor class for TextInputView
+     * class. There are some features added.
+     *
+     * - Invoke JavaScript's 'input' and 'change' event.
+     *
+     *   Note: React.js has only 'onChange' callback which has same behavior as
+     *   JavaScript's 'onInput' and does not have any callback corresponding to
+     *   original JavaScript's 'onChange'.
+     *   https://stackoverflow.com/questions/38256332/in-react-whats-the-difference-between-onchange-and-oninput
+     *
+     * - Support React's controlled component model.
+     *   https://reactjs.org/docs/uncontrolled-components.html
+     *   If field `controlled` is true,
+     *   only the string value given by `setControlledValue()`
+     *   is shown in the text editor.
+     *
+     * - Allow to set 'placeholder' text and 'placeholder-color' separately.
+     */
+    class TextInput : public juce::TextEditor, public juce::TextEditor::Listener
```

Comment:
I'd actually be tempted to make this a nested class in `TextView`. What do you think @nick-thompson ?

Reply:
On account of namespacing? Or actually, marking it as a private nested class makes sense to me, as this class interface is really a private implementation detail, it's not a user-facing API that we anticipate maintaining.

Yea, I'm :+1: for it

Reply:
I also think it's much better!
I will fix it.